js_modules/dagster-ui/packages/ui-components/src/components/MiddleTruncate.tsx

```diff
@@ -87,7 +87,9 @@ const Container = styled.div`
  */
 const calculateMiddleTruncatedText = (container: HTMLDivElement, text: string) => {
   const font = getComputedStyle(container).font;
-  const width = container.getBoundingClientRect().width;
+
+  // https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model/Determining_the_dimensions_of_elements#how_much_room_does_it_use_up
+  const width = container.offsetWidth;
```

Comment:
Not caused by your PR but can we wrap the usage of calculateMiddleTruncatedText in a `requestAnimationFrame` since accessing offsetWidth triggers a forced layout.

Reply:
Hmm it looks like this fn is called in two places - one in a useLayoutEffect and one in a resize observer. I think in the resizeObserver case the layout is fresh so it's best to read it immediately vs allowing to change and then requesting it again, but I'm not sure about the useLayoutEffect callsite.

It looks like using `useLayoutEffect` instead of `useEffect` is the [preferred way](https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen) to implement this without an empty frame where the text was `null` while we were waiting to determine it's size, but if we're ok with the empty frame + not forcing a render, i think we switch that to `useEffect` and add a `window.requestAnimationFrame`. 

I don't think it'll be that bad to have the `null` text for a single frame, I'll switch it and we can see if it's noticeable 👍

Reply:
Delay of doing it in a requestAnimationFrame isn't noticeable as far as I can tell - switched!